app/src/store/voice/call/callInvite.ts

```diff
@@ -0,0 +1,276 @@
+      return rejectWithValue({ reason: 'ID_NOT_FOUND_STATE' });
+    }
+
+    if (callInviteEntity.status !== 'idle') {
+      return rejectWithValue({ reason: 'STATUS_NOT_IDLE' });
+    }
+
+    dispatch(
+      setCallInvite({
+        action: 'accept',
+        id: callInviteEntity.id,
+        info: callInviteEntity.info,
+        status: 'pending',
+      }),
+    );
+
+    const acceptResult = await settlePromise(callInvite.accept());
+    if (acceptResult.status === 'rejected') {
+      return rejectWithValue({
+        reason: 'NATIVE_MODULE_REJECTED',
+        error: miniSerializeError(acceptResult.reason),
+      });
+    }
+
+    // dispatch(removeCallInvite(id));
+
+    const call = acceptResult.value;
+    const callInfo = getCallInfo(call);
+    return callInfo;
+  },
+);
+
+/**
+ * Reject call invite action.
+ */
+export type RejectCallInviteRejectValue =
+  | {
+      reason: 'ID_NOT_FOUND_MAP';
+    }
+  | {
+      reason: 'ID_NOT_FOUND_STATE';
+    }
+  | {
+      reason: 'STATUS_NOT_IDLE';
+    }
+  | {
+      reason: 'NATIVE_MODULE_REJECTED';
+      error: SerializedError;
+    };
+export const rejectActionTypes = generateThunkActionTypes(
+  `${sliceName}/reject`,
+);
+export const rejectCallInvite = createTypedAsyncThunk<
+  void,
+  { id: string },
+  { rejectValue: RejectCallInviteRejectValue }
+>(
+  rejectActionTypes.prefix,
+  async ({ id }, { dispatch, getState, rejectWithValue }) => {
+    const callInvite = callInviteMap.get(id);
+    if (typeof callInvite === 'undefined') {
+      return rejectWithValue({ reason: 'ID_NOT_FOUND_MAP' });
+    }
+
+    const callInviteEntity = getState().voice.call.callInvite.entities[id];
+    if (typeof callInviteEntity === 'undefined') {
+      return rejectWithValue({ reason: 'ID_NOT_FOUND_STATE' });
+    }
+
+    if (callInviteEntity.status !== 'idle') {
+      return rejectWithValue({ reason: 'STATUS_NOT_IDLE' });
+    }
+
+    dispatch(
+      setCallInvite({
+        action: 'reject',
+        id: callInviteEntity.id,
+        info: callInviteEntity.info,
+        status: 'pending',
+      }),
+    );
+
+    const rejectResult = await settlePromise(callInvite.reject());
+    if (rejectResult.status === 'rejected') {
+      return rejectWithValue({
+        reason: 'NATIVE_MODULE_REJECTED',
+        error: miniSerializeError(rejectResult.reason),
+      });
+    }
+
+    // dispatch(removeCallInvite(id));
+  },
+);
+
+/**
+ * Slice configuration.
+ */
+type BaseCallInviteEntity = {
+  id: string;
+  info: CallInviteInfo;
+};
+type CallInviteAcceptEntity = BaseCallInviteEntity &
+  AsyncStoreSlice<
+    { action: 'accept'; callInfo: CallInfo }, // fulfilled
+    { action: 'accept' } & (
+      | AcceptCallInviteRejectValue
+      | { error: SerializedError }
+    ), // rejected
+    { action: 'accept' } // pending
+  >;
+type CallInviteRejectEntity = BaseCallInviteEntity &
+  AsyncStoreSlice<
+    { action: 'reject' }, // fulfilled
```

Comment:
"fulfilled" for `{action: reject}` ?

Reply:
Ah, yeah I see the point of confusion. It's when we "reject" a CallInvite. But that action of rejecting is inherently asynchronous. So this line `192` is defining the fulfilled state of rejecting a CallInvite.